go/vt/vttablet/tabletserver/rules/rules_test.go

```diff
@@ -496,31 +497,68 @@ func TestAction(t *testing.T) {
 
 	bv := make(map[string]*querypb.BindVariable)
 	bv["a"] = sqltypes.Uint64BindVariable(0)
-	action, desc := qrs.GetAction("123", "user1", bv)
+
+	mc := sqlparser.MarginComments{
+		Leading:  "some comments leading the query",
+		Trailing: "other trailing comments",
+	}
+
+	action, desc := qrs.GetAction("123", "user1", bv, mc)
 	if action != QRFail {
 		t.Errorf("want fail")
 	}
 	if desc != "rule 1" {
 		t.Errorf("want rule 1, got %s", desc)
 	}
-	action, desc = qrs.GetAction("1234", "user", bv)
+	action, desc = qrs.GetAction("1234", "user", bv, mc)
 	if action != QRFailRetry {
 		t.Errorf("want fail_retry")
 	}
 	if desc != "rule 2" {
 		t.Errorf("want rule 2, got %s", desc)
 	}
-	action, _ = qrs.GetAction("1234", "user1", bv)
+	action, _ = qrs.GetAction("1234", "user1", bv, mc)
 	if action != QRContinue {
 		t.Errorf("want continue")
 	}
+
 	bv["a"] = sqltypes.Uint64BindVariable(1)
-	action, desc = qrs.GetAction("1234", "user1", bv)
+	action, desc = qrs.GetAction("1234", "user1", bv, mc)
 	if action != QRFail {
 		t.Errorf("want fail")
 	}
 	if desc != "rule 3" {
-		t.Errorf("want rule 2, got %s", desc)
+		t.Errorf("want rule 3, got %s", desc)
+	}
+
+	// reset bound variable 'a' to 0 so it doesn't match rule 3
+	bv["a"] = sqltypes.Uint64BindVariable(0)
+
+	qr4 := NewQueryRule("rule 4", "r4", QRFail)
+	qr4.SetTrailingCommentCond(".*trailing.*")
+
+	newQrs := qrs.Copy()
+	newQrs.Add(qr4)
+
+	action, desc = newQrs.GetAction("1234", "user1", bv, mc)
+	if action != QRFail {
+		t.Errorf("want fail")
+	}
+	if desc != "rule 4" {
+		t.Errorf("want rule 4, got %s", desc)
+	}
```

Comment:
change all the assertions with `asset,NotEqual` or other function checks over IF condition followed by `t.Error`

Reply:
sure thing -- I'll update the test cases I edit so that they'll have a consistent check style but will not expand to the full file if that seems reasonable?

Reply:
that's what I also do :+1:

Reply:
:white_check_mark: done